src/Tools/dotnet-dump/Dumper.cs

```diff
@@ -61,14 +61,11 @@ public async Task<int> Collect(IConsole console, int processId, string output, b
                 }
                 else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
                 {
-                    DiagnosticsHelpers.DumpType dumpType = type == DumpTypeOption.Heap ? DiagnosticsHelpers.DumpType.WithHeap : DiagnosticsHelpers.DumpType.Normal;
+                    var client = new DiagnosticsClient(processId);
+                    DumpType dumpType = type == DumpTypeOption.Heap ? DumpType.WithHeap : DumpType.Normal;
 
                     // Send the command to the runtime to initiate the core dump
-                    var hr = DiagnosticsHelpers.GenerateCoreDump(processId, output, dumpType, diag);
-                    if (hr != 0)
-                    {
-                        throw new InvalidOperationException($"Core dump generation FAILED 0x{hr:X8}");
-                    }
+                    client.WriteDump(dumpType, output, diag);
```

Comment:
What exception does WriteDump throw on error? Does that exception get catch in the below catch?

Reply:
Added it to the catch below. Thanks!